### PR TITLE
feat(proxy): Proxy exception list for outbound http connectors

### PR DESCRIPTION
**Pull request by johnBgood**
## Description

- REST connector can now use Java System Properties to configure a proxy (and nonProxyHosts as well).

## Related issues

closes [#3252](https://github.com/camunda/connectors/issues/3252)

## Checklist

- [x] PR has a **milestone** or the `no milestone` label.

### CODE REVIEW

**Review by mathias-vandaele — APPROVED** [review]
Good changes!

**Review by sbuettner — APPROVED** [review]
Awesome!

File: connectors/http/http-base/src/test/java/io/camunda/connector/http/base/client/apache/CustomApacheHttpClientProxyTest.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  private static GenericContainer<?> proxyContainer;
+
+  @BeforeAll
+  public static void setUp() {
+    proxyContainer =
+        new GenericContainer<>(DockerImageName.parse("sameersbn/squid:3.5.27-2"))
+            .withExposedPorts(3128)
+            .withClasspathResourceMapping("squid.conf", "/etc/squid/squid.conf", BindMode.READ_ONLY)
+            .waitingFor(org.testcontainers.containers.wait.strategy.Wait.forListeningPort());
+    proxyContainer.start();
```
Comment (sbuettner) [inline connectors/http/http-base/src/test/java/io/camunda/connector/http/base/client/apache/CustomApacheHttpClientProxyTest.java:62]: Great idea to use squid.

**Review by mathias-vandaele — APPROVED** [review]

**Comment by sbuettner** [comment]
Backport failed for `release/8.6`, because it was unable to cherry-pick the commit(s).

Please cherry-pick the changes locally and resolve any conflicts.
```bash
git fetch origin release/8.6
git worktree add -d .worktree/backport-3460-to-release/8.6 origin/release/8.6
cd .worktree/backport-3460-to-release/8.6
git switch --create backport-3460-to-release/8.6
git cherry-pick -x fb03259d191c8231f2e2cf59765ace267dc25229
```

**Pull request merged**